tests/test_integration.py

```diff
@@ -252,21 +252,35 @@ def test_send_presampler_headers_not_in_services(self, mocked_http):
         request_call = mocked_http.return_value.request
         assert request_call.call_count == 0
 
+    def _send_traces_and_check(self, traces, nresponses=1):
+        # test JSON encoder
+        responses = self.api_json.send_traces(traces)
+        assert len(responses) == nresponses
+        for response in responses:
+            assert response.status == 200
+
+        # test Msgpack encoder
+        responses = self.api_msgpack.send_traces(traces)
+        assert len(responses) == nresponses
+        for response in responses:
+            assert response.status == 200
+
     def test_send_single_trace(self):
         # register a single trace with a span and send them to the trace agent
         self.tracer.trace('client.testing').finish()
         trace = self.tracer.writer.pop()
         traces = [trace]
 
-        # test JSON encoder
-        response = self.api_json.send_traces(traces)
-        assert response
-        assert response.status == 200
+        self._send_traces_and_check(traces)
 
-        # test Msgpack encoder
-        response = self.api_msgpack.send_traces(traces)
-        assert response
-        assert response.status == 200
+    def test_send_many_traces(self):
+        # register a single trace with a span and send them to the trace agent
+        self.tracer.trace('client.testing').finish()
+        trace = self.tracer.writer.pop()
+        # 30k is a right number to have both json and msgpack send 2 payload :)
+        traces = [trace] * 30000
```

Comment:
This is likely fine, but could be brittle, if we add additional default metadata to a trace/span might make it be 3 or maybe we change encoding and it ends up all fitting in one!

It is a little longer, but we could determine the encoded size of this trace, and then if we know the max payload allowed size we could know exactly how many traces to include to make it be a given number of payloads.

This test case is likely fine right now as-is, but just pointing out we might need to adjust it later if any defaults change,